packages/ui/src/components/va-rating/hooks/useVaRatingColors.ts

```diff
@@ -0,0 +1,47 @@
+import { computed, ExtractPropTypes } from 'vue'
+import { useColors } from '../../../services/color-config/color-config'
+import { RatingValue } from '../types'
+
+export const useVaRatingColorsProps = {
+  unselectedColor: { type: String },
+  color: { type: String, default: 'primary' },
+  modelValue: { type: Number },
+}
+
+export const useVaRatingColors = (props: ExtractPropTypes<typeof useVaRatingColorsProps>) => {
+  const { getColor, getFocusColor, getTextColor } = useColors()
+
+  const computedColor = computed(() => getColor(props.color))
+
+  const backgroundColor = computed(() => {
+    if (props.unselectedColor) { return getColor(props.unselectedColor) }
+
+    return getFocusColor(getColor(props.color))
+  })
+
+  const backgroundComputed = computed(() => {
+    if (props.modelValue === RatingValue.HALF) {
+      return `linear-gradient(90deg, ${computedColor.value} 50%, ${backgroundColor.value} 50%`
+    }
+
+    if (props.modelValue === RatingValue.EMPTY) {
+      return backgroundColor.value
+    }
+
+    return computedColor.value
```

Comment:
A switch would be more appropriate

Reply:
I think "if" makes this much easier :) Usually switch is harder to read on two cases...